src/components/book/book.route.js

```diff
@@ -0,0 +1,36 @@
+import express from "express";
+import bookController from "./book.controller.js";
+import bookValidation from "./book.validation.js";
+import validation from "../../middleware/validation.js";
+import auth from "../../middleware/auth.js";
+
+const router = express.Router();
+
+router.post(
+  "/",
+  auth.userAuthenticate,
+  auth.accessRole(["admin"]),
+  validation.validate(bookValidation.addBookValidate),
+  bookController.addBook
+);
+
+router.put(
+  "/:id",
+  auth.userAuthenticate,
+  auth.accessRole(["admin"]),
+  validation.validate(bookValidation.addBookValidate),
+  bookController.updateBook
+);
+
+router.delete(
+  "/:id",
+  auth.userAuthenticate,
+  auth.accessRole(["admin"]),
+  bookController.RemoveBook
+);
+
+router.get("/search", bookController.searchBook);
```

Comment:
aa badhi APIs ma, compulsory ae to check karvu j pde ke user login che ke nai. te koi middleware nai nakhyu.

Reply:
naa km k user login vgr book to joi ske ne aevu rkhyu 6 me